master-firmware/src/base/map_server.cpp

```diff
@@ -118,3 +124,15 @@ void map_server_start(enum strat_color_t color)
     chThdCreateStatic(map_server_thd_wa, sizeof(map_server_thd_wa),
                       MAP_SERVER_PRIO, map_server_thd, &color);
 }
+
+struct _map* map_server_map_lock_and_get(void)
+{
+    chMtxLock(&map_data.map_lock);
+    return &map_data.map;
+}
+
+void map_server_map_release(struct _map* map_ptr)
+{
+    osalDbgAssert(map_ptr == &map_data.map, "wrong map ptr");
+    chMtxUnlock(&map_data.map_lock);
+}
```

Comment:
It's safer if you do the same with RAII, after all this is a `cpp` file :angel: 
```cpp
struct MapData
{
    MapData(void)
    {
        chMtxLock(&map_data.map_lock);
    }
    ~MapData(void)
    {
        chMtxUnlock(&map_data.map_lock);
    }
    operator struct _map*() const 
    {
        return &map_data.map;
    }
}
```

Reply:
yep yep, but those functions are called from C files... Otherwise of course it would have been RAII all the way :p

Reply:
time to make to `git mv strategy_helpers.c strategy_helpers.cpp`?